packages/default-reporter/src/reporterForClient/reportDeprecations.ts

```diff
@@ -1,18 +1,23 @@
 import { DeprecationLog } from '@pnpm/core-loggers'
 import * as Rx from 'rxjs'
-import { map } from 'rxjs/operators'
+import { filter, map } from 'rxjs/operators'
 import chalk from 'chalk'
 import formatWarn from './utils/formatWarn'
 import { zoomOut } from './utils/zooming'
+import type { LogLevel } from '@pnpm/logger'
 
 export default (
   deprecation$: Rx.Observable<DeprecationLog>,
   opts: {
     cwd: string
     isRecursive: boolean
+    logLevel?: LogLevel
   }
 ) => {
   return deprecation$.pipe(
+    // print deprecation warnings for all logLevel except error = warn info debug
+    // TODO unswitch -> return a "null pipe"
+    filter(() => opts.logLevel != 'error'),
```

Comment:
In this case, the stream should not be even read. Just return an empty stream instead, if loglevel is set to error.